api/src/main/java/org/openmrs/util/OpenmrsClassLoader.java

```diff
@@ -591,10 +591,7 @@ protected static void nullInstance(Object instance) {
 			}
 			try {
 				field.setAccessible(true);
-				if (Modifier.isStatic(mods) && Modifier.isFinal(mods)) {
-					// Doing something recursively is too risky
-					continue;
-				} else {
+				if(!Modifier.isStatic(mods) && !Modifier.isFinal(mods)) {
```

Comment:
this

`(!Modifier.isStatic(mods) && !Modifier.isFinal(mods))`

is not the else case of `(Modifier.isStatic(mods) && Modifier.isFinal(mods))`

this is

`(!Modifier.isStatic(mods) || !Modifier.isFinal(mods))`

since in the case of `Modifier.isStatic(mods)` being false `Modifier.isFinal(mods)` could still be true